devito/mpi/distributed.py

```diff
@@ -245,6 +245,12 @@ def nprocs(self):
     def topology(self):
         return self._topology
 
+    @cached_property
+    def boundary_rank(self):
```

Comment:
nitpicking (for homogeneity): in Devito, all introspection properties start with "is_"